core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -1312,8 +1308,13 @@ protected function getNewPosition($mode, $pid=null, $insertInto=false)
 					}
 				}
 
+				if (!$newPID)
+				{
+					$newPID = Widget::getEmptyValueByFieldType($GLOBALS['TL_DCA'][$this->strTable]['fields']['pid']['sql'] ?? array()) === null ? null : 0;
```

Comment:
I don‘t understand the ` === null ? null : 0` part. What else except `null` and `0` could possibly be returned for an INT() field?

Reply:
the widget class returns an empty string by default if there is no definition. This makes sure the value is either NULL or 0